lib/ui/common/ViewModelProviderBuilder.dart

```diff
@@ -18,18 +19,18 @@ typedef WidgetBuilder<T> = Widget Function(
     {BuildContext context, AsyncSnapshot<T> snapshot});
 
 class ViewModelProviderBuilder<T> extends StatelessWidget {
-  final ViewModelProvider<T> _provider;
+  final ViewModelProviderInterface<T> _defaultProvider;
```

Comment:
we keep the ability to directly pass a viewmodel provider, as that's handle for testing and playground